synapse/handlers/sync.py

```diff
@@ -818,6 +831,38 @@ def handle_room_entries(room_entry):
         defer.returnValue((newly_joined_rooms, newly_joined_users))
 
     @defer.inlineCallbacks
+    def _have_rooms_changed(self, sync_result_builder):
+        """Returns whether any rooms have changed since the sync. Must be an
+        incremental sync
```

Comment:
Returns true if there are any new events in the rooms the user is in since the last sync or if the user has changed membership in any room since the last sync.